lib/reline.rb

```diff
@@ -353,9 +353,11 @@ def readline(prompt = '', add_hist = false)
             line_editor.set_pasting_state(io_gate.in_pasting?)
             inputs.each do |key|
               if key.char == :bracketed_paste_start
+                line_editor.save_old_buffer
                 text = io_gate.read_bracketed_paste
                 line_editor.insert_pasted_text(text)
                 line_editor.scroll_into_view
+                line_editor.save_past_lines
```

Comment:
I think this save_old_buffer and save_past_lines can be hidden inside `insert_pasted_text`.

(Just a future idea, not related to this pull request)
Maybe we can change this part to handle bracketed paste in `line_editor.update(key)` like this in the future.
```ruby
inputs.each do |key|
  key.additional_data = io_gate.read_bracketed_paste if key.char == :bracketed_paste_start
  line_editor.update(key)
end
```